### PR TITLE
Jwt Authentication ADR

### PR DESCRIPTION
**Pull request by CDimonaco**
As the title suggest

### CODE REVIEW

File: adr/0007-use-jwt-tokens-as-authentication-mechanism.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+The new authentication solution should be based on standards and should be easy and appropriate to integrate into both Single Page Applications and raw API consumption.
+
+## Decision
+
+Decided on [JWT](https://jwt.io) token-based authentication.
+
+We don't want to use an API gateway and an external identity provider because the effort of deploying and mantaining these new pieces of architecture does not make sense right now.
+
+We want to use the web dashboard as an identity provider and authentication coordinator, this is a good compromise between the level of security we need and the type of application we are building right now. The choice of jwt as a token standard implies that a future switch to an external identity provider and/or API gateway will be a painless and easy procedure for our applications. 
+
```
Comment (fabriziosestito) [inline adr/0007-use-jwt-tokens-as-authentication-mechanism.md:25]: let's go a little bit more in detail with the considered alternatives if possible

File: adr/0007-use-jwt-tokens-as-authentication-mechanism.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+The new authentication solution should be based on standards and should be easy and appropriate to integrate into both Single Page Applications and raw API consumption.
+
+## Decision
+
+Decided on [JWT](https://jwt.io) token-based authentication.
+
+We don't want to use an API gateway and an external identity provider because the effort of deploying and mantaining these new pieces of architecture does not make sense right now.
+
+We want to use the web dashboard as an identity provider and authentication coordinator, this is a good compromise between the level of security we need and the type of application we are building right now. The choice of jwt as a token standard implies that a future switch to an external identity provider and/or API gateway will be a painless and easy procedure for our applications.
```
Comment (fabriziosestito) [inline adr/0007-use-jwt-tokens-as-authentication-mechanism.md:24]: jwt -> JWT

File: adr/0007-use-jwt-tokens-as-authentication-mechanism.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+## Decision
+
+Decided on [JWT](https://jwt.io) token-based authentication.
+
+We don't want to use an API gateway and an external identity provider because the effort of deploying and mantaining these new pieces of architecture does not make sense right now.
+
+We want to use the web dashboard as an identity provider and authentication coordinator, this is a good compromise between the level of security we need and the type of application we are building right now. The choice of jwt as a token standard implies that a future switch to an external identity provider and/or API gateway will be a painless and easy procedure for our applications. 
+
+The authentication procedure will be handled by [Pow](https://github.com/danschultzer/pow), the authentication library we **already** use on the web dashboard. 
+We change the underlying credentials dispatcher, to send the user a jwt access token, with a refresh jwt token for the refresh procedure.
```
Comment (fabriziosestito) [inline adr/0007-use-jwt-tokens-as-authentication-mechanism.md:27]: JWT

File: adr/0007-use-jwt-tokens-as-authentication-mechanism.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+## Consequences
+
+We remove the old authentication eex templates, served directly by Phoenix.
+The single page application will include a login page.
+The single page application will handle the access token authentication with the refresh flow.
+All the new services will use the jwt authentication mechanism
```
Comment (fabriziosestito) [inline adr/0007-use-jwt-tokens-as-authentication-mechanism.md:34]: JWT

**Review by dottorblaster — APPROVED** [review]
JWT casing apart, LGTM

**Review by nelsonkopliku — APPROVED** [review]
LGTM :rocket:

File: adr/0007-use-jwt-tokens-as-authentication-mechanism.md (hunk 1 above)
Comment (CDimonaco) [reply adr/0007-use-jwt-tokens-as-authentication-mechanism.md:27]: done

**Comment by CDimonaco** [comment]
Done all the nitpicks @fabriziosestito

**Review by fabriziosestito — APPROVED** [review]

**Pull request merged**